nvflare/app_opt/confidential_computing/cc_manager.py

```diff
@@ -0,0 +1,234 @@
+                currently supports the following content only
+
+                .. code-block:: json
+
+                    {
+                        "version":"1.0",
+                        "authorization-rules":{
+                            "x-nv-gpu-available":true,
+                            "x-nv-gpu-attestation-report-available":true,
+                            "x-nv-gpu-info-fetched":true,
+                            "x-nv-gpu-arch-check":true,
+                            "x-nv-gpu-root-cert-available":true,
+                            "x-nv-gpu-cert-chain-verified":true,
+                            "x-nv-gpu-ocsp-cert-chain-verified":true,
+                            "x-nv-gpu-ocsp-signature-verified":true,
+                            "x-nv-gpu-cert-ocsp-nonce-match":true,
+                            "x-nv-gpu-cert-check-complete":true,
+                            "x-nv-gpu-measurement-available":true,
+                            "x-nv-gpu-attestation-report-parsed":true,
+                            "x-nv-gpu-nonce-match":true,
+                            "x-nv-gpu-attestation-report-driver-version-match":true,
+                            "x-nv-gpu-attestation-report-vbios-version-match":true,
+                            "x-nv-gpu-attestation-report-verified":true,
+                            "x-nv-gpu-driver-rim-schema-fetched":true,
+                            "x-nv-gpu-driver-rim-schema-validated":true,
+                            "x-nv-gpu-driver-rim-cert-extracted":true,
+                            "x-nv-gpu-driver-rim-signature-verified":true,
+                            "x-nv-gpu-driver-rim-driver-measurements-available":true,
+                            "x-nv-gpu-driver-vbios-rim-fetched":true,
+                            "x-nv-gpu-vbios-rim-schema-validated":true,
+                            "x-nv-gpu-vbios-rim-cert-extracted":true,
+                            "x-nv-gpu-vbios-rim-signature-verified":true,
+                            "x-nv-gpu-vbios-rim-driver-measurements-available":true,
+                            "x-nv-gpu-vbios-index-conflict":true,
+                            "x-nv-gpu-measurements-match":true
+                        }
+                    }
+
+        """
+        FLComponent.__init__(self)
+        self.site_name = None
+        self.helper = None
+        self.verifiers = verifiers
+        self.my_token = None
+        self.participant_cc_info = {}  # used by the Server to keep tokens of all clients
+
+    def handle_event(self, event_type: str, fl_ctx: FLContext):
+        if event_type == EventType.SYSTEM_BOOTSTRAP:
+            try:
+                err = self._prepare_for_attestation(fl_ctx)
+            except:
+                self.log_exception(fl_ctx, "exception in attestation preparation")
+                err = "exception occurred"
+            if err:
+                self.log_critical(fl_ctx, err, fire_event=False)
+                sys.exit(-1)
```

Comment:
when handling the events, the handler should not call sys.exit(-1) during the error. You should just raise exception, for example: UnsafeComponentError.

Reply:
Got it.  Thanks.